src/h3/mod.rs

```diff
@@ -1838,40 +1866,55 @@ mod tests {
         s.send_body_client(stream2, true).unwrap();
         s.send_body_client(stream1, true).unwrap();
 
-        assert_eq!(s.poll_server(), Ok((stream1, Event::Headers(req1))));
-        assert_eq!(s.poll_server(), Ok((stream1, Event::Data)));
-        assert_eq!(s.recv_body_server(stream1, &mut recv_buf), Ok(body.len()));
-        assert_eq!(s.poll_server(), Ok((stream1, Event::Data)));
-        assert_eq!(s.recv_body_server(stream1, &mut recv_buf), Ok(body.len()));
-        assert_eq!(s.poll_server(), Ok((stream1, Event::Finished)));
-
-        assert_eq!(s.poll_server(), Ok((stream2, Event::Headers(req2))));
-        assert_eq!(s.poll_server(), Ok((stream2, Event::Data)));
-        assert_eq!(s.recv_body_server(stream2, &mut recv_buf), Ok(body.len()));
-        assert_eq!(s.poll_server(), Ok((stream2, Event::Data)));
-        assert_eq!(s.recv_body_server(stream2, &mut recv_buf), Ok(body.len()));
-        assert_eq!(s.poll_server(), Ok((stream2, Event::Finished)));
-
-        assert_eq!(s.poll_server(), Ok((stream3, Event::Headers(req3))));
-        assert_eq!(s.poll_server(), Ok((stream3, Event::Data)));
-        assert_eq!(s.recv_body_server(stream3, &mut recv_buf), Ok(body.len()));
-        assert_eq!(s.poll_server(), Ok((stream3, Event::Data)));
-        assert_eq!(s.recv_body_server(stream3, &mut recv_buf), Ok(body.len()));
-        assert_eq!(s.poll_server(), Ok((stream3, Event::Finished)));
+        let (stream, ev) = s.poll_server().unwrap();
```

Comment:
The test code is nicely adapted to the new functionality but it is quite duplicative. We might want to simplify this in a future PR; for example a loop or add a testing util function that can do the checks.

Reply:
Yeah, I turned this and the response checkes below into a loop.